tools/slicec-cs/src/decoding.rs

```diff
@@ -221,10 +221,25 @@ pub fn decode_sequence(sequence_ref: &TypeRef<Sequence>, namespace: &str, encodi
                 // We always read an array even when mapped to a collection, as it's expected to be
                 // faster than decoding the collection elements one by one.
                 Some(format!(
-                    "decoder.DecodeSequence<{}>()",
+                    "decoder.DecodeSequence<{}>({})",
                     element_type.cs_type_string(namespace, TypeContext::Decode, true),
+                    if matches!(primitive, Primitive::Bool) {
+                        r#"checkElement: element =>
+    {
+        if (System.Runtime.CompilerServices.Unsafe.As<bool, byte>(ref element) > 1)
```

Comment:
I would create a helper static method, e.g. `SliceDecoder.CheckBoolValue`, and simplify this generated code.